core/src/consensus/mod.rs

```diff
@@ -324,8 +328,10 @@ pub type PendingTransitionStore<'a> = Fn(H256) -> Option<PendingTransition> + 'a
 /// Voting errors.
 #[derive(Debug)]
 pub enum EngineError {
-    /// Signature or author field does not belong to an authority.
-    NotAuthorized(Address),
+    /// Precommit signatures or author field does not belong to an authority.
+    BlockNotAuthorized(Address),
+    /// The height or the signer index of message does not belong to an authority.
+    MessageNotAuthorized(u64, usize),
```

Comment:
How about adding `Option<Address>`?

Reply:
I've added `Option<Address>` on `MessageNotAuthorized`, and changed an error message.

Reply:
Spilt into `MessageNotAuthorized` and `ValidatorNotExist`.

Reply:
`MessageNotAuthorized` is changed to `MessageWithInvalidSignature`.